model/span.go

```diff
@@ -95,6 +94,31 @@ func (s *Span) NormalizeTimestamps() {
 	}
 }
 
+// ParentSpanID returns ID of a parent span if it exists.
+// It searches for the first child-of reference pointing to the same trace ID.
+func (s *Span) ParentSpanID() SpanID {
+	for i := range s.References {
```

Comment:
why not iterate over the reference directly? ie `for _, ref := `

Reply:
each iteration will have to copy 4 words into the on-stack var